### PR TITLE
Docs: fuel configurations section

### PR DESCRIPTION
**Pull request by camiinthisthang**
Thank you @SilentCicero for writing this up! I added definitions to each term at the top, please take a look at those and suggest any edits. 

One image is included in the configurations doc, this will be updated to have a transparent background so it looks better in the docs. Additionally, the photos you outlined in the doc are being worked on and will be added in a later PR.

### CODE REVIEW

File: src/SUMMARY.md [hunk 1]
```diff
@@ -13,11 +13,11 @@
...
     - [What is a Modular Execution Layer]()
-  - [Fuel Configurations]()
-    - [Fuel as an L1]()
-    - [Fuel as a rollup]()
-    - [Fuel as a statechannel]()
-    - [Fuel as a sidechain]()
+  - [Fuel Configurations](./configurations.md)
+    - [Fuel as an L1](./layer-1.md)
+    - [Fuel as a rollup](./rollup.md)
+    - [Fuel as a statechannel](./statechannel.md)
```
Comment (adlerjohn) [inline src/SUMMARY.md:21]: statechannel -> state channel everywhere

File: src/SUMMARY.md [hunk 2]
```diff
@@ -13,11 +13,11 @@
     - [Modular Blockchains]()
     - [Modular Blockchain Stack]()
     - [What is a Modular Execution Layer]()
-  - [Fuel Configurations]()
-    - [Fuel as an L1]()
-    - [Fuel as a rollup]()
-    - [Fuel as a statechannel]()
-    - [Fuel as a sidechain]()
+  - [Fuel Configurations](./configurations.md)
+    - [Fuel as an L1](./layer-1.md)
```
Comment (adlerjohn) [inline src/SUMMARY.md:19]: Hmmm. Should we re-order this to rollup first, then L1 second?

File: src/configurations.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Fuel Configurations
+
+Fuel is a modular architecture and is designed to operate in multiple different configurations.
+
+![a chart showing different Fuel configurations](https://user-images.githubusercontent.com/15346823/188051935-be751b92-5402-4216-aa2d-0e409fb356fd.png)
```
Comment (adlerjohn) [inline src/configurations.md:5]: Do you know what the persistence policy is for this? I think we should host this locally in the book (e.g. a relative link to an image file in some subdirectory).

Comment (camiinthisthang) [reply src/configurations.md:5]: true, will update this

**Pull request closed without merging**